source/OutfitterPanel.cpp

```diff
@@ -420,52 +424,65 @@ ShopPanel::BuyResult OutfitterPanel::CanBuy(bool onlyOwned) const
 
 		double engineNeeded = -selectedOutfit->Get("engine capacity");
 		double engineSpace = playerShip->Attributes().Get("engine capacity");
 		if(engineNeeded > engineSpace)
-			return "Only part of your ship's outfit capacity is usable for engines. "
+			addError("Only part of your ship's outfit capacity is usable for engines. "
 				"You cannot install this outfit, because it takes up "
 				+ Format::CargoString(engineNeeded, "engine space") + ", and this ship has "
-				+ Format::MassString(engineSpace) + " free.";
+				+ Format::MassString(engineSpace) + " free.");
 
 		if(selectedOutfit->Category() == "Ammunition")
-			return !playerShip->OutfitCount(selectedOutfit) ?
+			addError(!playerShip->OutfitCount(selectedOutfit) ?
 				"This outfit is ammunition for a weapon. "
 				"You cannot install it without first installing the appropriate weapon."
 				: "You already have the maximum amount of ammunition for this weapon. "
-				"If you want to install more ammunition, you must first install another of these weapons.";
+				"If you want to install more ammunition, you must first install another of these weapons.");
 
 		int mountsNeeded = -selectedOutfit->Get("turret mounts");
 		int mountsFree = playerShip->Attributes().Get("turret mounts");
 		if(mountsNeeded && !mountsFree)
-			return "This weapon is designed to be installed on a turret mount, "
-				"but your ship does not have any unused turret mounts available.";
+			addError("This weapon is designed to be installed on a turret mount, "
+				"but your ship does not have any unused turret mounts available.");
 
 		int gunsNeeded = -selectedOutfit->Get("gun ports");
 		int gunsFree = playerShip->Attributes().Get("gun ports");
 		if(gunsNeeded && !gunsFree)
-			return "This weapon is designed to be installed in a gun port, "
-				"but your ship does not have any unused gun ports available.";
+			addError("This weapon is designed to be installed in a gun port, "
+				"but your ship does not have any unused gun ports available.");
 
 		if(selectedOutfit->Get("installable") < 0.)
-			return "This item is not an outfit that can be installed in a ship.";
+			addError("This item is not an outfit that can be installed in a ship.");
 
 		// For unhandled outfit requirements, show a catch-all error message.
-		return "You cannot install this outfit in your ship, "
+		if(errors.empty())
+		addError("You cannot install this outfit in your ship, "
 			"because it would reduce one of your ship's attributes to a negative amount. "
-			"For example, it may use up more cargo space than you have left.";
+			"For example, it may use up more cargo space than you have left.");
+	}
+
+	if(errors.empty())
+		return true;
+	else if(errors.size() == 1)
+		return errors[0];
+	else
+	{
+		string errorMessage = "There are several reasons why you cannot buy this outfit:\n";
+		for(size_t i = 0; i < errors.size(); ++i)
+			errorMessage += to_string(i + 1) + ". " + errors[i] + "\n";
```

Comment:
We already have a similar listing in the game (see https://github.com/endless-sky/endless-sky/pull/9189#issuecomment-1700679161), so maybe this should look more like that one.

Reply:
I agree the UI should be consistent here, I will make the necessary changes. 
While playing the game this week, I noticed that when you land on a planet that completes multiple missions, the outcome of each mission is displayed one at a time in its own window. It might be beneficial to revise this to display them all in one window for consistency, with the gross total earned displayed at the bottom.

Reply:
> While playing the game this week, I noticed that when you land on a planet that completes multiple missions, the outcome of each mission is displayed one at a time in its own window. It might be beneficial to revise this to display them all in one window for consistency, with the gross total earned displayed at the bottom.

I'm not really a fan of this, but feel free to open a separate issue/PR if you want to discuss it.

Reply:
re my comment above, after this PR is accepted, I'm willing to make the changes to the 'mission complete window' if you think it's a good idea.

Reply:
Okay, noted, I don't feel passionately about it.